src/components/cutscenes/RivalRoute22LeavesCutscene.ts

```diff
@@ -0,0 +1,71 @@
+import { Component } from "eightbittr/lib/Component";
+
+import { FullScreenPokemon } from "../../FullScreenPokemon";
+import { IWalkingInstructions } from "../actions/Walking";
+import { Direction } from "../Constants";
+import { ICharacter } from "../Things";
+
+/**
+ * OakIntroRivalLeaves cutscene functions used by FullScreenPokemon instances.
+ */
+export class RivalRoute22LeavesCutscene<TGameStartr extends FullScreenPokemon> extends Component<TGameStartr> {
+    /**
+     * Cutscene for determing if the walking scene should be run.
+     */
+    public AfterBattle(): void {
+        if (this.gameStarter.battles.isPartyWiped) {
```

Comment:
A tip for testing code: whenever you see an `if` statement, that represents two code paths you need to test. There's the `true` case and the `false` case. Test both.